activiti-cloud-audit-service/activiti-cloud-services-audit-liquibase/src/main/java/org/activiti/cloud/services/audit/liquibase/ActivitiCloudAuditLiquibaseAutoConfiguration.java

```diff
@@ -0,0 +1,76 @@
+ * limitations under the License.
+ */
+package org.activiti.cloud.services.audit.liquibase;
+
+import javax.sql.DataSource;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.liquibase.LiquibaseAutoConfiguration;
+import org.springframework.boot.autoconfigure.liquibase.LiquibaseProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.ResourceLoaderAware;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.io.ResourceLoader;
+
+import liquibase.exception.LiquibaseException;
+import liquibase.integration.spring.SpringLiquibase;
+
+@Configuration
+@ConditionalOnProperty(prefix = "spring.liquibase", name = "enabled", matchIfMissing = true)
+@PropertySource("classpath:config/audit-liquibase.properties")
+public class ActivitiCloudAuditLiquibaseAutoConfiguration implements ResourceLoaderAware {
+
+    private ResourceLoader resourceLoader;
+
+    @Override
+    public void setResourceLoader(ResourceLoader resourceLoader) {
+        this.resourceLoader = resourceLoader;
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(name = "auditLiquibase")
+    public SpringLiquibase auditLiquibase(DataSource dataSource) throws LiquibaseException {
+        return buildSpringLiquibase(dataSource, auditLiquibaseProperties());
+    }
+
+    @Bean
+    @ConfigurationProperties(prefix = "spring.audit.liquibase")
+    public LiquibaseProperties auditLiquibaseProperties() {
+        return new LiquibaseProperties();
+    }
+
+    private SpringLiquibase buildSpringLiquibase(DataSource dataSource, LiquibaseProperties properties) {
+        SpringLiquibase liquibase = new SpringLiquibase();
+        liquibase.setDataSource(dataSource);
+        liquibase.setResourceLoader(resourceLoader);
+        liquibase.setChangeLog(properties.getChangeLog());
+        liquibase.setContexts(properties.getContexts());
+        liquibase.setDefaultSchema(properties.getDefaultSchema());
+        liquibase.setLiquibaseTablespace(properties.getLiquibaseTablespace());
+        liquibase.setDropFirst(properties.isDropFirst());
+        liquibase.setShouldRun(properties.isEnabled());
+        liquibase.setLabels(properties.getLabels());
+        liquibase.setChangeLogParameters(properties.getParameters());
+        liquibase.setRollbackFile(properties.getRollbackFile());
+        liquibase.setDatabaseChangeLogTable(properties.getDatabaseChangeLogTable());
+        liquibase.setDatabaseChangeLogLockTable(properties.getDatabaseChangeLogLockTable());
+        liquibase.setTestRollbackOnUpdate(properties.isTestRollbackOnUpdate());
+        return liquibase;
+    }
```

Comment:
We should consider extracting this block in a shared place, as we have exactly the same code for `audit`, `query`, and `modeling`.

Reply:
Fixed in 6ea8111da574263878d73a62104037772f0be3f9